### PR TITLE
Developer Guide Update

### PR DESCRIPTION
**Pull request by Brainsucker92**
#### Short Description
Updates the developer guide at http://opentosca.github.io/container/DeveloperGuide.html

#### Proposed Changes

  * Fixes serveral typos
  * Update guide to latest project version
  * Grammatical corrections

**Fixes**: #

### CODE REVIEW

**Comment by Brainsucker92** [comment]
Quick comment to https://github.com/OpenTOSCA/container/pull/63/commits/9cc7c2da36011cec877862751fb6f9496f8021c2 : 
Eclipse showed me this issue as 

> Cannot resolve element with id 'createprovide-a-declarative-service'


error in my IDE. It's very likely that is actually not a real issue, since the shortcut is working perfectly fine on the Dev guide page. @koppor already assumed this might be an issue caused by Eclipse itself, while parsing the markdown. In case this is true, we have to revert this change.

Eclipse is also showing another similar error:
>Cannot resolve element with id 'consume--use-a-declarative-service'

which occurs just one line below the other one. It seems like there is an issue with parsing the '/' in the markdown.
Depending on the result of the other issue we might have to fix this one as well or leave it as it is.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -210,30 +209,31 @@ Fig. 6 shows the correct project structure of the Core File Service used as exam
 
 A service can be only consumed in a plug-in project.
 
-1. Open the bundle manifest "**META-INF/MANIFEST.MF**" (relative to project root) of the plug-in
+1. Create new "Plug-in Project" called "**org.opentosca.core.file.service.consumer**"
+2. Open the bundle manifest "**META-INF/MANIFEST.MF**" (relative to project root) of the plug-in
    project that should consume a service.
-2. In the appearing Eclipse bundle manifest editor go to the tab "**Dependencies**"
-3. At "**Imported Packages**" click on "**Add...**", select the package that includes the interface of the needed service an click on "**OK**".
+3. In the appearing Eclipse bundle manifest editor go to the tab "**Dependencies**"
```
Comment (koppor) [inline docs/DeveloperGuide.md]: No need to adapt all numbers. Markdown takes care at rendering. If you update, you better change all numbers to `1.`.

Comment (Brainsucker92) [reply docs/DeveloperGuide.md]: Thanks for the hint. I assume, we can leave it as it is now, right?

**Comment by koppor** [comment]
Regarding the question on the non-resolved ids. You did not change the markdown:

grafik

So, one can open http://opentosca.github.io/container/DeveloperGuide to check the anchor.

Oh, wow, the *CURRENT* homepage works!!!!

click

It uses this URL: http://opentosca.github.io/container/DeveloperGuide#createprovide-a-declarative-service

Do you see that in your browser, too?

Why did you change the target URL?

If I open http://opentosca.github.io/container/DeveloperGuide.html#create-provide-a-declarative-service in my browser, it DOES NOT jump to the heading "Create/provide a declarative service".

**Comment by Brainsucker92** [comment]
The lines I was taling about were lines 23 and 24, which look in the old version in Eclipse like this: 

error

This is why I thought some corrections were necessary. But actually I think I have to revert this change, since it appears to be working on the page.

**Comment by miwurster** [comment]
@Brainsucker92 What tasks/work is left here?

**Comment by scheffsn-dev** [comment]
@miwurster Nothing I can think of, actually.
Maybe @nyuuyn got any idea, if there's someting to add/change.
If not, this can be merged.

**Pull request merged**